### PR TITLE
GAPI: Overload for `GArray<GMat>` ProtoParam

### PR DESCRIPTION
**Pull request by OrestChura**
1) `GArray<GMat>` ProtoParam's type in `gtyped.hpp` is now `std::vector<cv::Mat>`

It is needed for typed version of `GComputation`,  `GComputationT` in case a graph's input/output node is `GArray<GMat>`

2) Tests for using GArray<GMat> with GComputation added

3) Added the possibility to pass compile_args to GComputationT::apply() fro convenient usage in tests

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2020.3.0:16.04
build_image:Custom Win=openvino-2020.3.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by OrestChura** [comment]
Waited for builds :) @dmatveev plz take a look

File: modules/gapi/include/opencv2/gapi/gtyped.hpp [hunk 1]
```diff
@@ -26,6 +26,7 @@ namespace detail
     template<> struct ProtoToParam<cv::GMat>    { using type = cv::Mat; };
     template<> struct ProtoToParam<cv::GScalar> { using type = cv::Scalar; };
     template<typename U> struct ProtoToParam<cv::GArray<U> >  { using type = std::vector<U>; };
+    template<> struct ProtoToParam<cv::GArray<cv::GMat>>      { using type = std::vector<cv::Mat>; };
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/gtyped.hpp:29]: maybe it is woth aligning by ProtoToParam<> too.

**Comment by OrestChura** [comment]
> Can you please write a test that this construct will really work now with our typed API?

Sure, will update soon

File: modules/gapi/include/opencv2/gapi/gtyped.hpp (hunk 1 above)
Comment (OrestChura) [reply modules/gapi/include/opencv2/gapi/gtyped.hpp:29]: I decided not to align like that because the line will be too long otherwise

**Comment by OrestChura** [comment]
To reason my changes:
I had to add compile_args functionality for `GComputationT.apply()` because otherwise it wouldn't be so easy to write tests with custom kernels
And unfortunately windows didn't like the implementation with default parameter `compile_args = {}` (it couldn't "find" the proper function and failed with "C2660: this function can't have `n` arguments") so I just added the overload with `compile_args`

**Comment by OrestChura** [comment]
@dmatveev please take a look on tests

File: modules/gapi/include/opencv2/gapi/gtyped.hpp (hunk 1 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gtyped.hpp:29]: ok

**Comment by asmorkalov** [comment]
@mshabunin Please take a look and merge.

**Pull request merged**